Backend/src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,102 @@
+import { Request, Response } from 'express';
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+import { User, UserInput } from '../models/user.model';
+import { Token } from '../models/token.model';
+
+const generateToken = (userId: string, secret: string, expiresIn: string) => {
+  return jwt.sign({ id: userId }, secret, { expiresIn });
+};
+
+const register = async (req: Request, res: Response) => {
+  const { email, password, fullName, role } = req.body;
+  if (!email || !fullName || !password) {
+    return res.status(422).json({ message: 'The fields email, fullNamea and password are required' });
+  }
+
+  try {
+    const user = await User.findOne({ email });
+    if (user) {
+      return res.status(400).send('User already exist');
```

Comment:
Can we get these HTTP status codes from some constants? Usually libraries have these values defined already as constants.

Reply:
HTTP status codes are defined in a separate constants file and referenced from there.

Reply:
Ok